### PR TITLE
New Mecha Stuff

### PR DESCRIPTION
**Pull request by Kurfursten**
That mech painting PR renewed my interest in mecha stuff. Adding some new gear.

**Powered Exosuit Running Board**
* Makes your entry into the exosuit instant
* Only attaches to working class mechs (Ripley, Clarke)
* Available from robotics

**Exosuit-Mounted Rack**
* Acts like a large bag you can access by clicking the mech (if DNA locked, only the owner can access it)
* Can pull items in and out with the help of a powerful new storage proc for interacting with storage objects inside something else
* Available from robotics

**Exosuit Modkit**
* Increases the number of equipment attachments your mecha can hold, once
* Available via Trader

---

Pretty thoroughly tested.

🆑 
* rscadd: Added 2 new mecha tools (rack and runningboard) to robotics and 1 mecha upgrade to the trader.

### CODE REVIEW

File: code/game/objects/items/trader.dm [hunk 1]
```diff
@@ -752,3 +752,37 @@
...
+		return
+	to_chat(user,"<span class='notice'>You begin modifying the exosuit.</span>")
+	working = TRUE
+	if(do_after(user,target,4 SECONDS))
+		to_chat(user,"<span class='notice'>You finish modifying the exosuit!</span>")
+		M.max_equip++
+		qdel(src)
+	else
+		to_chat(user,"<span class='notice'>You stop modifying the exosuit.</span>")
+		in_use = FALSE
```
Comment (boy2mantwicethefam) [inline code/game/objects/items/trader.dm]: you might want to do working = FALSE here

File: code/game/objects/items/trader.dm [hunk 2]
```diff
@@ -752,3 +752,37 @@
...
+
+/obj/item/weapon/mech_expansion_kit
+	name = "exosuit expansion kit"
+	desc = "All the equipment you need to replace that useless legroom with a useful bonus equipment slot on your mech."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "modkit"
+	flags = FPRINT
+	siemens_coefficient = 0
+	w_class = W_CLASS_SMALL
+	var/working = FALSE
```
Comment (boy2mantwicethefam) [inline code/game/objects/items/trader.dm:764]: Is this really necessary? The do_after should fail if the item in question disappears so you really can't upgrade the mecha several times with the same item

File: code/game/objects/items/trader.dm (hunk 1 above)
Comment (Kurfursten) [reply code/game/objects/items/trader.dm]: good catch

File: code/game/objects/items/trader.dm (hunk 2 above)
Comment (Kurfursten) [reply code/game/objects/items/trader.dm:764]: it isn't necessary but it allows me to have a special fail message

File: code/game/mecha/equipment/passive/passive.dm [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	max_combined_w_class = 28
+	slot_flags = 0
+	rustle_sound = "rustle-metal"
+	var/obj/item/mecha_parts/mecha_equipment/mech_part
+
+/obj/item/weapon/storage/mechrack/Destroy()
+	mech_part = null
+	..()
+
+/obj/item/weapon/storage/mechrack/distance_interact(mob/user)
```
Comment (boy2mantwicethefam) [inline code/game/mecha/equipment/passive/passive.dm:41]: Try testing it with telekinesis, just to be sure

Comment (Kurfursten) [reply code/game/mecha/equipment/passive/passive.dm:41]: I actually factored this in already, but your comment made me realize that on line 44 it should be checking the range of `user,M`, not `src,M`, so good thought

**Comment by Kurfursten** [comment]
Tested with TK just to be sure. It has the peculiar effect of allowing you to view the mecha's rack inventory from a range [if you have access], but you can't take things in or out. I think this is fine.

**Review by ShiftyRail — APPROVED** [review]
Don't see anything particularly shocking here

File: code/game/mecha/mecha.dm [hunk 4]
```diff
@@ -1392,11 +1394,14 @@
...
 
 /obj/mecha/proc/operation_allowed(mob/living/carbon/human/H)
+	if(dna)
+		if(!(usr.dna.unique_enzymes==dna))
+			return FALSE
 	if(istype(H))
 		for(var/ID in list(H.get_active_hand(), H.wear_id, H.belt))
-			if(src.check_access(ID,src.operation_req_access))
+			if(src.check_access(ID,operation_req_access))
 				return 1
```
Comment (jknpj) [inline code/game/mecha/mecha.dm:1403]: return 1 and return FALSE? :thinking:

**Comment by dunsel-hordon** [comment]
Did this break the equipment icon that would pop up in the top right? Was in a ripley and no equipment icon was popping up.

**Comment by Kurfursten** [comment]
The passive equipment is not supposed to show a spell icon, that's a feature

**Comment by dunsel-hordon** [comment]
no i couldn't see active equipment either like my scythe or clamp. Maybe it was just a byond issue.

**Pull request merged**